text/dotnet-core/rfcs/0000-debug-support.md

```diff
@@ -0,0 +1,79 @@
+# Enable Remote Debug Support
+
+## Proposal
+
+The .NET Core buildpack should enable remote debugging of applications in a
+manner that conforms to [Project
+RFC0037](https://github.com/paketo-buildpacks/rfcs/blob/main/text/0037-remote-debug.md).
+To accomplish this, the buildpack will need to install the Visual Studio
+Debugger into the application run image. The debugger can attach to a running
+`dotnet` process and be bound to a client-side debugger via STDIN across a
+connection invoked via `docker exec` or `kubectl exec`.
+
+## Motivation
+
+Remote debugging is a useful practice for understanding complex program logic
+and interactions in remote environments including production. Providing this
+feature for .NET Core developers will bring this buildpack into alignment with
+the types of features we already provide for developers in other language
+families like Java.
+
+## Implementation
+
+Implementing this feature will require changes in 2 existing buildpacks,
+`dotnet-publish` and `dotnet-execute`, and the addition of a new buildpack,
+`vsdbg`, to the `dotnet-core` language-family buildpack.
+
+### `vsdbg`
+
+This buildpack will be developed to enable the installation of the Visual
```

Comment:
Once installed, will the `vsdbg` debugger just know how to attach to the `dotnet` process at launch time? Or will we need to also include some attachment config as outlined in https://github.com/Microsoft/MIEngine/wiki/Offroad-Debugging-of-.NET-Core-on-Linux---OSX-from-Visual-Studio#attaching?

Reply:
That configuration happens on the IDE side. It should probably default to PID 1 though since that's where the `dotnet` process should be running.